pkg/scaffold/v2/main.go

```diff
@@ -79,7 +79,15 @@ func main() {
 
 	ctrl.SetLogger(zap.Logger(true))
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{ Scheme: scheme, MetricsBindAddress: metricsAddr })
+	// Persmissions to do leader election.
+	// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+	// +kubebuilder:rbac:groups="",resources=configmaps/status,verbs=get;update;patch
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                  scheme,
+		MetricsBindAddress:      metricsAddr,
+		LeaderElection:          true,
+		LeaderElectionNamespace: "default",
```

Comment:
"default" namespace. Can we parameterize this and supply the namespace where deployment itself is deployed. I think we can supply using downward API ?

Reply:
I believe `DownwardAPI` will work here.